pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -38,6 +38,8 @@ def connect(self, model: torch.nn.Module, optimizers, lr_schedulers):
         """Connects the precision plugin to the training process,
         configures apex and reinits the schedulers
         """
+        if model.device.type != "cuda":
+            return model, optimizers, lr_schedulers
```

Comment:
We should never get here if the device type is CUDA right?

Reply:
this is only needed, because in Spawn the main process (which isn't the local rank 0 here) actually calls setup as well, but without a model being on gpu, meaning that we shouldn't do anythin then.
We cannot move it to gpu before, since we then occupy memory on GPU0 in the main proc and the actual rank 0.